sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java

```diff
@@ -408,9 +409,25 @@ private static void fieldDescriptorFromTableField(
       case "INT64":
       case "INTEGER":
         if (value instanceof String) {
-          return Long.valueOf((String) value);
+          try {
+            return Long.valueOf((String) value);
+          } catch (NumberFormatException e) {
+            // Expected. Element will be added to the failed element PCollection
```

Comment:
These are going to result in a generic `SchemaDoesntMatchException("Unexpected value :"...)` being thrown which will make this potentially difficult to debug when it happens. If we can't just let the exception bubble up here, you should wrap it with a new exception rather than just dropping it. (Same thing for the exceptions below.)

Reply:
Addressed in the latest commit. Only implemented for the INTEGER type at the moment.